src/image.cpp

```diff
@@ -95,6 +95,7 @@ namespace {
         AccessMode     commentSupport_;
     };
 
+    /// \todo Use std::unordered_map for implementing the registry. Avoid to use ImageType::none
```

Comment:
`std::array` is imho better here, each element has a unique id and these ids are sequential starting at zero. There's not really a better datastructure for this than an array.

Reply:
Only if you know size at compile time.

Reply:
We do, the index is `ImageType`.

Reply:
It depends. If we want to conserve a relationship between integer values and ImageTypes, then yes. But is this something we should do? I do not think we need to do it ...

As I commented previously, I do not think this change will have an impact on users since those ImageType values are used mostly internally. I do not see how a user would want to rely in those values in external software. Therefore, I think that `std::unordered_map` would be a better option with `ImageType` as key, an a Register instance as value. The order of the `ImageType` would not matter for the Registry implementation in that way, and we could keep that scope enum sorted alphabetically. 

Anyways, I just left the comment there to address this later, and we could discuss about the best way to implement that in other issue/PR :wink: